citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/parse/iterators/CrossRefIterator.java

```diff
@@ -0,0 +1,270 @@
+        List<Reference> references = new ArrayList<>();
+        if(itemReferences != null) {
+            for(edu.asu.diging.crossref.model.Reference itemRef: itemReferences) { 
+                Reference ref = new Reference();
+                ref.setAuthorString(itemRef.getAuthor());
+                ref.setContributors(null);
+                ref.setTitle(itemRef.getArticleTitle());
+                ref.setYear(itemRef.getYear());
+                if(itemRef.getDoi()!=null && !itemRef.getDoi().isBlank()) {
+                    ref.setIdentifier(itemRef.getDoi());
+                    ref.setIdentifierType("DOI");
+                    ref.setSource(itemRef.getDoiAssertedBy());
+                } else if (itemRef.getIssn()!=null && !itemRef.getIssn().isBlank()) {
+                    ref.setIdentifier(itemRef.getIssn());
+                    ref.setIdentifierType("ISSN");
+                } else if (itemRef.getIsbn()!=null && !itemRef.getIsbn().isBlank()) {
+                    ref.setIdentifier(itemRef.getIsbn());
+                    ref.setIdentifierType("ISBN");
+                }
+                ref.setFirstPage(itemRef.getFirstPage());
+                ref.setVolume(itemRef.getVolume());
+                ref.setReferenceId(itemRef.getKey());
+                ref.setReferenceString(itemRef.getUnstructured());
+                ref.setReferenceStringRaw(itemRef.getUnstructured());
+                references.add(ref);
+            }
+        }
+        return references;
+    }
+
+    private List<Contributor> mapPersonToContributor(List<Person> personList) {
+        List<Contributor> contributors = new ArrayList<Contributor>();
+        for(Person person: personList) {
+            Contributor contributor = new Contributor();
+            contributor.setContributionType(ContributionType.EDITOR);
+            contributor.setGivenName(person.getGiven());
+            contributor.setSurname(person.getFamily());
+            contributor.setFullName(person.getName());
+            List<Affiliation> affiliations = new ArrayList<>();
+            for(Institution institute: person.getAffiliation()) {
+                Affiliation affiliation = new Affiliation();
+                affiliation.setName(institute.getName());
+                affiliations.add(affiliation);
+            }
+            contributor.setAffiliations(affiliations);
+            ContributorId contributorID = new ContributorId();
+            contributorID.setId(person.getOrcid());
+            contributorID.setIdSystem("ORCID");
+            contributor.setIds(Arrays.asList(contributorID));
+            contributors.add(contributor); 
+        }
+        return contributors;
+    }
+
+    @Override
+    public BibEntry next() {
+        if (iteratorDone) {
+            return null;
+        }
+        BibEntry nextEntry = new Publication();;
+        
+        try {
+            Item item = crossrefService.get(info.getDois().get(currentIndex));
```

Comment:
instead of keeping an int around, just get an interator from the dois list and keep that